File: solutions/typescript/resistor-color/1/resistor-color.ts
export const colorCode = (color: string): number => {
  try {
    return COLORS.indexOf(color)
  } catch (error) {
    throw new Error(`Provided color ${color} does not exist.`)
  }
}

export const COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white'
]

File: solutions/python/resistor-color/1/resistor_color.py
def color_code(color: str) -> int:
    """Gets the corresponding number of given color

    :param color: str - color to get the number of
    :return: int - color's number code
    """
    return colors().index(color)


def colors() -> list[str]:
    """Returns all the colors in order

    :return: list[str] - all the colors in order
    """
    return [
        "black",
        "brown",
        "red",
        "orange",
        "yellow",
        "green",
        "blue",
        "violet",
        "grey",
        "white",
    ]
